pspmutating.go

```diff
@@ -0,0 +1,133 @@
+			if err != nil {
+				return false, diff, err
+			}
+			parentPod = ds.Spec.Template
+		}
+		if owner.Kind == "Node" {
+			return false, diff, fmt.Errorf("Pod with ownerReference of kind Node is not supported. OwnerReference of pod %v was %#v", pod.Name, owner)
+		}
+		if diffNew := deep.Equal(GetContainerSecurityContexts(parentPod.Spec), GetContainerSecurityContexts(pod.Spec)); diffNew != nil {
+			diff = append(diff, diffNew...)
+		}
+		if diffNew := deep.Equal(parentPod.Spec.SecurityContext, pod.Spec.SecurityContext); diffNew != nil {
+			diff = append(diff, diffNew...)
+		}
+		if diffNew := deep.Equal(GetPSPAnnotations(parentPod.ObjectMeta.Annotations), GetPSPAnnotations(pod.ObjectMeta.Annotations)); diffNew != nil {
+			diff = append(diff, diffNew...)
+		}
+	}
+	if len(diff) > 0 {
+		return true, diff, nil
+	}
+	return false, diff, nil
+}
+
+// IsPSPMutating checks wheter a PodSecurityPolicy is potentially mutating
+// pods. It returns true if one of the fields or annotations used in the
+// PodSecurityPolicy is suspected to be mutating pods. The field or annotations
+// that are suspected to be mutating are returned as well.
+func IsPSPMutating(pspObj *v1beta1.PodSecurityPolicy) (mutating bool, fields, annotations []string) {
+	fields = make([]string, 0)
+	annotations = make([]string, 0)
+
+	if len(pspObj.Spec.DefaultAddCapabilities) > 0 {
+		fields = append(fields, "DefaultAddCapabilities")
+	}
+	if len(pspObj.Spec.RequiredDropCapabilities) > 0 {
+		fields = append(fields, "RequiredDropCapabilities")
+	}
+	if pspObj.Spec.SELinux.Rule != v1beta1.SELinuxStrategyRunAsAny {
+		fields = append(fields, "SELinux")
+	}
+	if pspObj.Spec.RunAsUser.Rule != v1beta1.RunAsUserStrategyRunAsAny {
+		fields = append(fields, "RunAsUser")
+	}
+	if pspObj.Spec.RunAsGroup != nil && pspObj.Spec.RunAsGroup.Rule == v1beta1.RunAsGroupStrategyMustRunAs {
+		fields = append(fields, "RunAsGroup")
+	}
+	if pspObj.Spec.SupplementalGroups.Rule != v1beta1.SupplementalGroupsStrategyRunAsAny {
+		fields = append(fields, "SupplementalGroups")
+	}
+	if pspObj.Spec.FSGroup.Rule != v1beta1.FSGroupStrategyRunAsAny {
+		fields = append(fields, "FSGroup")
+	}
+	if pspObj.Spec.ReadOnlyRootFilesystem != false {
+		fields = append(fields, "ReadOnlyRootFilesystem")
+	}
+	if pspObj.Spec.DefaultAllowPrivilegeEscalation != nil {
+		fields = append(fields, "DefaultAllowPrivilegeEscalation")
+	}
+	if pspObj.Spec.AllowPrivilegeEscalation != nil && *pspObj.Spec.AllowPrivilegeEscalation != true {
+		fields = append(fields, "AllowPrivilegeEscalation")
+	}
+
+	mutatingAnnotations := make(map[string]bool)
+	mutatingAnnotations["seccomp.security.alpha.kubernetes.io/defaultProfileName"] = true
+	mutatingAnnotations["apparmor.security.beta.kubernetes.io/defaultProfileName"] = true
+
+	for k, _ := range pspObj.Annotations {
+		if _, ok := mutatingAnnotations[k]; ok {
+			annotations = append(annotations, k)
+		}
+	}
```

Comment:
nit: prefer
```suggestion
	mutatingAnnotations := []string{
		"seccomp.security.alpha.kubernetes.io/defaultProfileName",
		"apparmor.security.beta.kubernetes.io/defaultProfileName",
	}
	for _, a := range mutatingAnnotations {
		if _, ok := pspObj.Annotations[k]; ok {
			annotations = append(annotations, k)
		}
	}
```

Reply:
not sure why my brain was originally thinking in inverse.... fixed this.